tempesta_fw/http.c

```diff
@@ -3162,6 +3162,16 @@ tfw_http_resp_cache_cb(TfwHttpMsg *msg)
 	TfwHttpReq *req = resp->req;
 
 	TFW_DBG2("%s: req = %p, resp = %p\n", __func__, req, resp);
+	/*
+	 * Response will never be sent: client has disconnected already.
+	 * Don't adjust response, simply destroy it.
+	 */
+	if (unlikely(test_bit(TFW_HTTP_B_REQ_DROP, req->flags))) {
+		TFW_DBG2("%s: resp=[%p] dropped: client disconnected\n",
+			 __func__, resp);
+		tfw_http_resp_pair_free(req);
+		return;
+	}
```

Comment:
A client can be disconnected at any time: before we execute a caching process (also not so lightweight) or while the response is in forwarding queue, so it's not obvious why did you choose exactly this place. However, the place looks good - it makes sense to store a response in a cache regardless the client connection, so other clients can hit the cache entry. If this is what you had in mind, then just describe it in the comment above. Otherwise let's have a discussion on this in the chat.

Reply:
Yes, that was the exact reason. Should have commented it explicitly.